airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/format/format_command.py

```diff
@@ -75,16 +75,53 @@ def get_help_message(self) -> str:
             message = f"{message}."
         return message
 
-    def get_dir_to_format(self, dagger_client: dagger.Client) -> dagger.Directory:
-        """Get the directory to format according to the file_filter.
+    async def get_dir_to_format(self, dagger_client) -> Directory:
+        """Get a directory with all the source code to format according to the file_filter.
+        We mount the the files to format in a git container and remove all gitignored files.
+        It ensures we're not formatting files that are gitignored.
 
         Args:
-            dagger_client (dagger.Client): The dagger client to use to get the directory
+            dagger_client (dagger.Client): The dagger client to use to get the directory.
 
         Returns:
-            dagger.Directory: The directory to format
+            Directory: The directory with the files to format that are not gitignored.
         """
-        return dagger_client.host().directory(self.LOCAL_REPO_PATH, include=self.file_filter, exclude=DEFAULT_FORMAT_IGNORE_LIST)
+        # Load a directory from the host with all the files to format according to the file_filter and the .gitignore files
+        dir_to_format = dagger_client.host().directory(
+            self.LOCAL_REPO_PATH, include=self.file_filter + ["**/.gitignore", ".gitignore"], exclude=DEFAULT_FORMAT_IGNORE_LIST
```

Comment:
Isn't `[".gitignore"]` enough? If not, `["**/.gitignore"]` will be.

Reply:
Yep, sorry it was a manual testing legacy.